src/wallet/wallet.ts

```diff
@@ -143,16 +180,45 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
    * @memberof HathorWalletServiceWallet
    * @inner
    */
-  async start() {
+  async start({ pinCode }) {
     if (!this.seed) {
       throw new Error('Seed should be in memory when starting the wallet.');
     }
 
     this.setState(walletState.LOADING);
+
+    const xpriv = walletUtils.getXPrivKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
     const xpub = walletUtils.getXPubKeyFromSeed(this.seed, {passphrase: this.passphrase, networkName: this.network.name});
+    const authXpub = HathorWalletServiceWallet.getAuthXPubKeyFromSeed(this.seed, {
+      passphrase: this.passphrase,
+      networkName: this.network.name,
+    });
+
+    const walletId = walletUtils.sha256d(xpub, 'hex');
+    const now = new Date();
+    const timestampNow = Math.floor(now.getTime() / 1000);
+
+    // prove we own the xpubkey
+    const xpubAccountPath = walletUtils.deriveXpriv(xpriv, '0\'')
+    const address = xpubAccountPath.publicKey.toAddress(this.network.getNetwork()).toString();
+    const message = new bitcore.Message(String(timestampNow).concat(walletId).concat(address));
```

Comment:
Should we make the message creation as a static method on `HathorWalletServiceWallet` that receive the timestamp, walletId and address?

Reply:
I think we don't need it to be `static` as it is only being used on this class, I refactored this part of the code on 715e5f9abc085093f6180fd809d86959c1e7a10b

Can you please check again?